src/runtime_src/core/common/detail/linux/sysinfo.h

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// Local - Include files
+#include "core/common/sysinfo.h"
+
+// 3rd Party Library - Include Files
+#include <boost/property_tree/ptree.hpp>
+#include <boost/format.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+
+// System - Include Files
+#include <gnu/libc-version.h>
+#include <sys/utsname.h>
+#include <thread>
+
+#if defined(__aarch64__) || defined(__arm__) || defined(__mips__)
+  #define MACHINE_NODE_PATH "/proc/device-tree/model"
+#elif defined(__PPC64__)
+  #define MACHINE_NODE_PATH "/proc/device-tree/model-name"
+  // /proc/device-tree/system-id may be 000000
+  // /proc/device-tree/model may be 00000
+#elif defined (__x86_64__)
+  #define MACHINE_NODE_PATH "/sys/devices/virtual/dmi/id/product_name"
+#else
+#error "Unsupported platform"
+  #define MACHINE_NODE_PATH ""
+#endif
+
+namespace {
+static std::string 
+machine_info()
+{
+  std::string model("unknown");
+  std::ifstream stream(MACHINE_NODE_PATH);
+  if (stream.good())
+    std::getline(stream, model);
+  return model;
+}
+
+static boost::property_tree::ptree
+glibc_info()
+{
+  boost::property_tree::ptree _pt;
+  _pt.put("name", "glibc");
+  _pt.put("version", gnu_get_libc_version());
+  return _pt;
+}
+
+} //end anonymous namespace
+
+
+namespace xrt_core::sysinfo {
```

Comment:
This namespace should be `xrt_core::sysinfo::detail`.   Your core/common/sysinfo.h should declare `xrt_core::sysinfo::get_os_info()`.  So `core/common/sysinfo.cpp` should include `core/common/detail/sysinfo.h` and sysinfo.cpp should implement `xrt_core::sysinfo::get_os_info()` to call `xrt_core::sysinfo::detail::get_os_info()`

Reply:
I see, this makes sense. Thank you!